inc/saidebugcounter.h

```diff
@@ -0,0 +1,277 @@
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saidebugcounter.h
+ *
+ * @brief   This module defines SAI Debug Counter interface
+ *
+ * @par Abstract
+ *
+ *    This module defines SAI Debug Counter API.
+ */
+
+#if !defined (__SAIDEBUGCOUNTER_H_)
+#define __SAIDEBUGCOUNTER_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIDEBUGCOUNTER SAI - Debug counter specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Debug counter type
+ */
+typedef enum _sai_debug_counter_type_t
+{
+    /** Port in drop reasons. Base object: SAI_OBJECT_TYPE_PORT */
+    SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS,
+
+    /** Port out drop reasons. Base object: SAI_OBJECT_TYPE_PORT */
+    SAI_DEBUG_COUNTER_TYPE_PORT_OUT_DROP_REASONS,
+
+} sai_debug_counter_type_t;
+
+/**
+ * @brief Debug counter bind method
+ */
+typedef enum _sai_debug_counter_bind_method_t
+{
+    /** Bind automatically to all instances of base object */
+    SAI_DEBUG_COUNTER_BIND_METHOD_AUTOMATIC,
+
+} sai_debug_counter_bind_method_t;
+
+/**
+ * @brief Attribute data for port in drop reasons
+ */
+typedef enum _sai_port_in_drop_reason_t
+{
+    /* L2 reasons */
+
+    /** Source MAC is multicast */
+    SAI_PORT_IN_DROP_REASON_SMAC_MULTICAST,
+
+    /** Source MAC equals Destination MAC */
+    SAI_PORT_IN_DROP_REASON_SMAC_EQUALS_DMAC,
+
+    /** Destination MAC is Reserved (Destination MAC=01-80-C2-00-00-0x) */
+    SAI_PORT_IN_DROP_REASON_DMAC_RESERVED,
+
+    /**
+     * @brief VLAN tag not allowed
+     *
+     * Frame tagged when port is dropping tagged,
+     * or untagged when dropping untagged
+     */
+    SAI_PORT_IN_DROP_REASON_VLAN_TAG_NOT_ALLOWED,
+
+    /** Ingress VLAN filter */
+    SAI_PORT_IN_DROP_REASON_INGRESS_VLAN_FILTER,
+
+    /** Ingress STP filter */
+    SAI_PORT_IN_DROP_REASON_INGRESS_STP_FILTER,
+
+    /** Unicast FDB table action discard */
+    SAI_PORT_IN_DROP_REASON_FDB_UC_DISCARD,
+
+    /** Multicast FDB table empty tx list */
+    SAI_PORT_IN_DROP_REASON_FDB_MC_DISCARD,
+
+    /** Port loopback filter */
+    SAI_PORT_IN_DROP_REASON_LOOPBACK_FILTER,
+
+    /* L3 reasons */
+
+    /** IPv4 Unicast Destination IP is link local (Destination IP=169.254.0.0/16) */
+    SAI_PORT_IN_DROP_REASON_DIP_LINK_LOCAL,
+
+    /** IPv4 Source IP is link local (Source IP=169.254.0.0/16) */
+    SAI_PORT_IN_DROP_REASON_SIP_LINK_LOCAL,
+
+    /** Packet size is larger than the MTU */
+    SAI_PORT_IN_DROP_REASON_EXCEEDS_MTU,
+
+    /* ACL reasons */
+
+    /** Packet is dropped due to configured ACL rules */
+    SAI_PORT_IN_DROP_REASON_ACL_DISCARD
+
+} sai_port_in_drop_reason_t;
+
+/**
+ * @brief Attribute data for port out drop reasons
+ */
+typedef enum _sai_port_out_drop_reason_t
+{
+    /** Egress VLAN filter */
+    SAI_PORT_OUT_DROP_REASON_EGRESS_VLAN_FILTER,
+
+    /** Packet is destined for neighboring device but neighbor device link is down */
+    SAI_PORT_OUT_DROP_REASON_L3_EGRESS_LINK_DOWN,
```

Comment:
can you add SAI_PORT_IN_DROP_REASON_L3_EGRESS_LINK_DOWN?

Reply:
It is already there - in egress reasons - 
    /** Packet is destined for neighboring device but neighbor device link is down */
    SAI_PORT_OUT_DROP_REASON_L3_EGRESS_LINK_DOWN,

Reply:
I believe he's asking to have it included as both an egress reason and an ingress reason, as different platforms could implement this behavior as an ingress counter or an egress counter (or both).